waterfall/java/com/google/waterfall/usb/AndroidUsbModule.java

```diff
@@ -0,0 +1,41 @@
+package com.google.waterfall.usb;
+
+import static android.content.Context.USB_SERVICE;
+
+import android.content.Context;
+import android.hardware.usb.UsbManager;
+import androidx.annotation.VisibleForTesting;
+import dagger.Module;
+import dagger.Provides;
+import javax.inject.Singleton;
+
+@Module
+public class AndroidUsbModule {
+
+  private final UsbService service;
+  @VisibleForTesting static UsbManagerIntf usbManager;
```

Comment:
I haven't used Dagger in a while, but wouldn't you create a separate module for testing? In that testing module your provideUsbManger implementation would return your test implementation.

Reply:
The problem is that we need to have the same object available in the instrumentation and target context since this is stubbing out binder calls. If I let dagger do its thing the manager in the test and the manager in the target will be two different instances.